File: typescript-client-sdk/src/sdk/models/operations/getallcrmeventattendees.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetAllCrmEventAttendeesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;
}


export class GetAllCrmEventAttendeesResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=eventAttendees", elemType: shared.EventAttendee })
  eventAttendees?: shared.EventAttendee[];

  @SpeakeasyMetadata({ data: "json, name=nextPageCursor" })
  nextPageCursor?: string;
}


export class GetAllCrmEventAttendeesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetAllCrmEventAttendeesQueryParams;
}


export class GetAllCrmEventAttendeesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetAllCrmEventAttendeesResponseBody;
}

File: go-client-sdk/pkg/models/operations/getallcrmeventattendees.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetAllCrmEventAttendeesQueryParams struct {
	AccessToken *string  `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool    `queryParam:"style=form,explode=true,name=allFields"`
	Cursor      *string  `queryParam:"style=form,explode=true,name=cursor"`
	Limit       *float64 `queryParam:"style=form,explode=true,name=limit"`
}

type GetAllCrmEventAttendeesResponseBody struct {
	EventAttendees []shared.EventAttendee `json:"eventAttendees,omitempty"`
	NextPageCursor *string                `json:"nextPageCursor,omitempty"`
}

type GetAllCrmEventAttendeesRequest struct {
	QueryParams GetAllCrmEventAttendeesQueryParams
}

type GetAllCrmEventAttendeesResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetAllCrmEventAttendeesResponseBody
}
